src/openrct2-ui/windows/Map.cpp

```diff
@@ -136,1455 +128,1279 @@ static constexpr const ScreenCoordsXY MiniMapOffsets[] = {
-    switch (get_current_rotation())
+    try
     {
-        case 0:
-            x = _currentLine * COORDS_XY_STEP;
-            y = 0;
-            dx = 0;
-            dy = COORDS_XY_STEP;
-            break;
-        case 1:
-            x = MAXIMUM_TILE_START_XY;
-            y = _currentLine * COORDS_XY_STEP;
-            dx = -COORDS_XY_STEP;
-            dy = 0;
-            break;
-        case 2:
-            x = MAXIMUM_MAP_SIZE_BIG - ((_currentLine + 1) * COORDS_XY_STEP);
-            y = MAXIMUM_TILE_START_XY;
-            dx = 0;
-            dy = -COORDS_XY_STEP;
-            break;
-        case 3:
-            x = 0;
-            y = MAXIMUM_MAP_SIZE_BIG - ((_currentLine + 1) * COORDS_XY_STEP);
-            dx = COORDS_XY_STEP;
-            dy = 0;
-            break;
+        rct_window* w = WindowFocusOrCreate<MapWindow>(WC_MAP, 245, 259, WF_10);
+        w->selected_tab = 0;
+        w->list_information_type = 0;
+        return w;
     }
-
-    for (int32_t i = 0; i < MAXIMUM_MAP_SIZE_TECHNICAL; i++)
+    catch (const std::bad_alloc&)
     {
-        if (!map_is_edge({ x, y }))
-        {
-            switch (w->selected_tab)
-            {
-                case PAGE_PEEPS:
-                    colour = MapWindowGetPixelColourPeep({ x, y });
-                    break;
-                case PAGE_RIDES:
-                    colour = MapWindowGetPixelColourRide({ x, y });
-                    break;
-            }
-            destination[0] = (colour >> 8) & 0xFF;
-            destination[1] = colour;
-        }
-        x += dx;
-        y += dy;
-
-        destinationPosition.x++;
-        destinationPosition.y++;
-        destination = _mapImageData.data() + (destinationPosition.y * MAP_WINDOW_MAP_SIZE) + destinationPosition.x;
+        return nullptr;
     }
-    _currentLine++;
-    if (_currentLine >= MAXIMUM_MAP_SIZE_TECHNICAL)
-        _currentLine = 0;
 }
 
-static CoordsXY MapWindowScreenToMap(ScreenCoordsXY screenCoords)
+void WindowMapReset()
 {
-    screenCoords.x = ((screenCoords.x + 8) - MAXIMUM_MAP_SIZE_TECHNICAL) / 2;
-    screenCoords.y = ((screenCoords.y + 8)) / 2;
-    auto location = TileCoordsXY(screenCoords.y - screenCoords.x, screenCoords.x + screenCoords.y).ToCoordsXY();
+    rct_window* w;
 
-    switch (get_current_rotation())
+    // Check if window is even opened
+    w = window_bring_to_front_by_class(WC_MAP);
+    if (w == nullptr)
     {
-        case 0:
-            return location;
-        case 1:
-            return { MAXIMUM_MAP_SIZE_BIG - 1 - location.y, location.x };
-        case 2:
-            return { MAXIMUM_MAP_SIZE_BIG - 1 - location.x, MAXIMUM_MAP_SIZE_BIG - 1 - location.y };
-        case 3:
-            return { location.y, MAXIMUM_MAP_SIZE_BIG - 1 - location.x };
+        return;
     }
 
-    return { 0, 0 }; // unreachable
+    auto* mapWindow = dynamic_cast<MapWindow*>(w);
```

Comment:
this doesn't need to be a dynamic cast can just be static (i think). WC_MAP is only set on MapWindow's so it is safe to call after using a bring_to_front.

Reply:
You're right, thanks! I'm still getting used to c++ casts.